### PR TITLE
docs: update archetype integration and x-adapter docs

### PR DESCRIPTION
**Pull request by herrardo**
EX-6728

Update documentation about archetype integration, archetype development and x-adapter

# Pull request template
 

- Updated infomation about env=live|staging as the archetype is failing when we use live as environment in the snippet.
- Updated examples about environment
- Updated type=module in a couple of examples 
- Bring Luisa's changes from docs portal to be matching her updates, but not everywhere, because we had more detailed information in some cases. This has been done checking this PR to update de documentation in docs that will be declined:  https://github.com/empathyco/docs-empathy-documentation/pull/609/files

## Motivation and context
Better docs for partners and customers, and updated info about x-adapter

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link:EX-6728

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [x] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ attribute `src` so that it points to the staging environment as follows:
 <script>
   window.initX = {
     instance: 'my-store',
-    env: 'live', // Note that here you are using a production API with the staging version of Interface X
```
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:134]: I would keep this message either here or in another place. It is a point that I found users struggle. They don't realise that they can load the staging x-components with the production API or viceversa.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 2]
```diff
@@ -83,7 +82,7 @@ window.initX = {
 window.initX = function () {
   return {
     instance: 'my-store',
-    env: location.href.includes('.pre.') ? 'staging' : 'live',
+    ...(location.href.includes('.pre.') ? { env: 'staging' } : {}),
```
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:85]: Just sending `undefined` works and it is a simpler technique for the users to understand (I guess). We still have to deal with the env selector in our demo though. It doesn't accept `undefined` values, and there are some endpoints that trigger a 400 when used in production.

```suggestion
    env: location.href.includes('.pre.') ? 'staging' : undefined,
```

**Review by javieri-empathy — APPROVED** [review]

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 3]
```diff
@@ -224,7 +225,7 @@ following configuration options:
 | Name                                                   | Type                                                                                 | Description                                                                                                                                                                                                                                                                                      |
```
Comment (mlcorton) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:225]: Name of what? What do you really mean: Name or Parameter?

[the remark this answers is not in the document]
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:225]: You can include a new column to indicate the default values (if any) or the supported values. Also, you can indicate the supported values (especially for boolean types) in the description field.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md [hunk 4]
```diff
@@ -297,7 +297,8 @@ etc.). See the corresponding `events.types.ts` file for each module in the
 The
 [X&nbsp;API](https://github.com/empathyco/x/blob/main/packages/x-components/src/x-installer/api/base-api.ts)
 object allows your commerce store to communicate with Interface&nbsp;X. It supports multiple
-functions to integrate Interface&nbsp;X in your website.
+functions to integrate Interface&nbsp;X in your website. You may access these functions inside the
```
Comment (mlcorton) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:300]:
```suggestion
functions to integrate Interface&nbsp;X in your website. You can access these functions inside the
```

**Review by mlcorton — COMMENTED** [review]
I believe there are some inconsistencies in the snippet config table. Please, review my comment just in case I'm right and you need to apply any changes :)

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:225]: All of those are types in Typescript Luisa, this is correct.

A type could be a union of values: `'white' | 'black'` or an array of strings `string[]` or a Record... `any` is also a type that allows anything

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:225]: Name of the parameter...haha, we may change it to `Paramater`

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: Changed to parameter, as we has in old docs, makes sense

[the remark this answers is not in the document]
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: Touché! I've been reviewing JSON files for so long I'm afraid 😭

**Review by mlcorton — APPROVED** [review]
Well done, guys! 💪🏼

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**